src/tike/communicators/pool.py

```diff
@@ -132,11 +210,43 @@ def reduce_mean(self, x: list, axis, worker=None) -> cp.array:
             axis=axis,
         )
 
+    def grouped_allreduce(self, x: list, s: int):
```

Comment:
Call this function "all_reduce" to be consistent with the other functions that do not have "grouped" prefix. A default value of s = len(x) would make it the conventional all_reduce?